### PR TITLE
Toolpad-app dependencies update

### PR DESCRIPTION
**Pull request by Janpot**

### CODE REVIEW

**Review by apedroferreira — APPROVED** [review]

File: packages/toolpad-core/package.json [hunk 1]
```diff
@@ -33,8 +33,8 @@
     "url": "https://github.com/mui/mui-toolpad/issues"
   },
   "dependencies": {
-    "@mui/x-data-grid-pro": "^5.15.2",
-    "@tanstack/react-query": "^4.1.0",
+    "@mui/x-data-grid-pro": "^5.17.1",
```
Comment (apedroferreira) [inline packages/toolpad-core/package.json:36]: @Janpot i think this broke the datagrids

Comment (apedroferreira) [reply packages/toolpad-core/package.json:36]: we should add some integration test to see if datagrid work sometime soon

Comment (Janpot) [reply packages/toolpad-core/package.json:36]: In what sense? They seemed to work for me. Can you open a ticket?

Comment (apedroferreira) [reply packages/toolpad-core/package.json:36]: i've created an issue for the bug, hopefully it's not just some local issue i'm having but didn't seem like it
bug: https://github.com/mui/mui-toolpad/issues/921
integration test issue: https://github.com/mui/mui-toolpad/issues/920

**Pull request merged**